### ISSUE TITLE
SQL dependency doesn't track query

### ISSUE THREAD

**Issue by damienpontifex** [Microsoft/ApplicationInsights-dotnet#427]
We are using app insights with ASP.NET MVC targeting .NET 4.6.1, using AppInsights 2.2.0 and data access through EntityFramework.

When viewing SQL dependencies in app insights through the azure portal, our actual query isn't being logged. Currently the only information we get is

Command
"tcp:_servername_.database.windows.net,1433 | _databasename_"

Is there something we're missing or can check with configuring app insights so that the actual query also gets logged? 

Thanks in advance for any assistance.

**Comment by cijothomas** [comment]
@damienpontifex SQL query is collected if Profiler is installed either via Status Monitor or ApplicationInsights extension in Azure.

**Comment by damienpontifex** [comment]
I have now installed the extension on my web app on azure. But still no query logged into the Dependency window under Command...

Also, if it is required to install the extension, I feel like that's a bit of a UX breakdown in the portal if there isn't any help text like 'Want to see the full query, install the extension into your web app' to allow us to discover this solution

**Comment by SergeyKanzhelev** [comment]
@iusafaro can you help here?

**Comment by iusafaro** [comment]
1) Have you restarted web app after extension installation?
	2) Do you see informational event in Application event log from “.NET Runtime”, ID=1022, Message=“.NET Runtime version 4.0.30319.34209 - The profiler was loaded successfully.  Profiler CLSID: '{324F817A-7420-4E6D-B3C1-143FBED6D855}'.”? It’s written during your app launch i.e. new w3wp instance created. 
	Eventlog.xml is located at: Web App resource menu -> Advanced Tools -> Go -> Debug Console (command bar in the header) -> LogFiles. New events are at the very bottom.
	3) Check ApplicationInsightsExtension.log from the same folder - does it have "End of Application Insights Profiler configuration" without any exceptions before this message?
	4) Go to Analytics portal and run "dependencies | where type == "SQL" and timestamp >= ago(10m) | take 1" check sdkVersion of returned item and it should start with rddp in case of Profiler was successfully enabled and rddf if it wasn't.

**Comment by SergeyKanzhelev** [comment]
@damienpontifex I'm closing this bug for now. Please follow instructions provided by @iusafaro and let us know the results for further troubleshooting

**Comment by RickBlouch** [comment]
This might be obvious, but it's not to me and I couldn't figure it out after googling and searching the web.  How do I "install ApplicationInsights extension in Azure" for a service fabric application running .net core?  

I ran the query you listed in step 4 above and it is returning rddf so I know that it's not set up properly.

My .net core app is has the Microsoft.ApplicationInsights.AspNetCore 2.0.0 Nuget package.  Like the original post, I am seeing data in the app insights, just not the EF generated SQL.

Thanks.

**Comment by dnduffy** [comment]
@RickBlouch For .NET Core that support is coming in the next release.  Please post on http://github.com/Microsoft/ApplicationInsights-aspnetcore for questions regarding the AI SDK for .NET Core.

**Comment by Fluwit** [comment]
I cannot get this to work in a Service Fabric environment with a non-Core service either. It has never worked in ASF, in fact. Currently I'm considering writing a EF interceptor to do this kind of logging, but that seems silly since this is supposed to work?

I'm using the v2.3.0-beta1 by the way.

**Comment by bragma** [comment]
Hi, what are the performance implications of installing the Profiler? Can it significantly impact a production website? Thanks.

**Comment by cijothomas** [comment]
@bragma if you are already using latest version of SDK, adding profiler should not have much difference.

**Comment by johnkwaters** [comment]
I could use some help too here. I can see which Web API calls take the longest, and was expecting
a) To for these see what SQL is generated in each DB query
b) See a list somewhere of longest/most frequent SQL queries, just like you can for API calls.
Where to I find this?
I have the profiler running, I am in an App Service, using .NET Fwk 4.7 and AI 2.4

**Comment by cijothomas** [comment]
@johnkwaters 
a -Installing application insights extension will collect SQL queries. 
b- You can use analytics queries to find answers. (https://docs.microsoft.com/en-us/azure/application-insights/app-insights-analytics) I don't think the UI provides grouping etc based on SQL queries, so analytics is the best bet.

**Comment by duongthaiha** [comment]
Hi 
We use service fabric on prem to host our services. I have seen extension of IIS Server and WebApp etc however there is nothing for Service Fabric. Is there one available or is there another way for us to see the query please?

**Comment by kfrajtak** [comment]
Hi,
I have a Win32 desktop .net application and I am not able to make AI to track the SQL dependency. I have added the `Microsoft.ApplicationInsights.DependencyCollector` Nuget package (v2.4.0.) which modified the `ApplicationsInsights.config` file:

```
<ApplicationInsights xmlns="http://schemas.microsoft.com/ApplicationInsights/2013/Settings">
  <!-- 
    Learn more about Application Insights configuration with ApplicationInsights.config here: 
    http://go.microsoft.com/fwlink/?LinkID=513840        
  -->
  <TelemetryChannel>
    <DeveloperMode>true</DeveloperMode>
  </TelemetryChannel>
  <TelemetryModules>
    <Add Type="Microsoft.ApplicationInsights.DependencyCollector.DependencyTrackingTelemetryModule, Microsoft.AI.DependencyCollector">
      <ExcludeComponentCorrelationHttpHeadersOnDomains>       
        <Add>core.windows.net</Add>
        <Add>core.chinacloudapi.cn</Add>
        <Add>core.cloudapi.de</Add>
        <Add>core.usgovcloudapi.net</Add>
      </ExcludeComponentCorrelationHttpHeadersOnDomains>
    </Add>
  </TelemetryModules>
  ...
```

I even tried the "manual" configuration:
```
DependencyTrackingTelemetryModule dependencyTracking = new DependencyTrackingTelemetryModule();
dependencyTracking.Initialize(TelemetryConfiguration.Active);
```

The app is using a custom ORM which is using `System.Data.dll`.

Can you please give me a hint what might be wrong?

Thank you,
Kare;

**Comment by khoanguyen** [comment]
I got the same problem and found this thread https://social.msdn.microsoft.com/Forums/en-US/11423d8a-a065-439c-82fa-9832ea392c5e/app-insights-not-logging-ef-620-queries?forum=ApplicationInsights, it explains why EF queries were not logged to me.

Using EF 6.4, I implemented a DBCommand interceptor for manual tracking.
IDbCommandInterceptor only available in EF 6.2 and later

```
    public class EntityFrameworkAITracking : IDbCommandInterceptor
    {
        private static TelemetryClient _telemetryClient = new TelemetryClient();

        [ThreadStatic]
        private static Stopwatch _sw;

        public void NonQueryExecuted(DbCommand command, DbCommandInterceptionContext<int> interceptionContext)
        {
            StopTracking(command, interceptionContext);
        }

        public void NonQueryExecuting(DbCommand command, DbCommandInterceptionContext<int> interceptionContext)
        {
            StartTracking();
        }

        public void ReaderExecuted(DbCommand command, DbCommandInterceptionContext<DbDataReader> interceptionContext)
        {
            StopTracking(command, interceptionContext);
        }

        public void ReaderExecuting(DbCommand command, DbCommandInterceptionContext<DbDataReader> interceptionContext)
        {
            StartTracking();
        }

        public void ScalarExecuted(DbCommand command, DbCommandInterceptionContext<object> interceptionContext)
        {
            StopTracking(command, interceptionContext);
        }

        public void ScalarExecuting(DbCommand command, DbCommandInterceptionContext<object> interceptionContext)
        {
            StartTracking();
        }

        private void StartTracking()
        {
            _sw = Stopwatch.StartNew();
        }

        private void StopTracking<T>(DbCommand command, DbCommandInterceptionContext<T> interceptionContext)
        {
            _sw.Stop();
            var startTime = DateTimeOffset.UtcNow.Subtract(_sw.Elapsed);
            var target = $"{command.Connection.DataSource} | {command.Connection.Database}";
            var success = interceptionContext.Exception == null;
            var exception = interceptionContext.Exception;
            var data = command.CommandText;
            var resultCode = 0;

            if (!success)
            {
                if (exception is SqlException)
                {
                    resultCode = (exception as SqlException).Number;
                }

                _telemetryClient.TrackException(exception);
            }
            
            _telemetryClient.TrackDependency("SQL",
                target,
                $"EF: {target}",
                data,
                startTime,
                _sw.Elapsed,
                resultCode.ToString(),
                success);
        }
    }
```
